tests/integration/tier0/non-destructive/kernel/test_custom_kernel.py

```diff
@@ -107,17 +107,19 @@ def test_custom_kernel(convert2rhel, shell, custom_kernel, hybrid_rocky_image):
         os_vendor = "Rocky"
 
     if os.environ["TMT_REBOOT_COUNT"] == "1":
-        with convert2rhel("--debug") as c2r:
-            # We need to get past the data collection acknowledgement.
-            c2r.expect("Continue with the system conversion?")
-            c2r.sendline("y")
+        try:
+            with convert2rhel("--debug") as c2r:
+                # We need to get past the data collection acknowledgement.
+                c2r.expect("Continue with the system conversion?")
+                c2r.sendline("y")
 
-            c2r.expect("WARNING - Custom kernel detected. The booted kernel needs to be signed by {}".format(os_vendor))
-            c2r.expect("RHEL_COMPATIBLE_KERNEL::INVALID_KERNEL_PACKAGE_SIGNATURE")
+                c2r.expect(
+                    "WARNING - Custom kernel detected. The booted kernel needs to be signed by {}".format(os_vendor)
+                )
+                c2r.expect("RHEL_COMPATIBLE_KERNEL::INVALID_KERNEL_PACKAGE_SIGNATURE")
 
-            c2r.sendcontrol("c")
+                c2r.sendcontrol("c")
 
-        try:
             assert c2r.exitstatus == 1
-        except AssertionError:
+        except:  # pylint: disable=W0702
```

Comment:
It still may be too broad, but maybe we should use Exception here, or try to see what is the base exception we could catch

Reply:
We started with an AssertionError on the c2r.exitcode, but then I realized, that anything going wrong during the contextmanager call would not be handled that way.
I felt that even though it's too broad, it might be more reliable.
Generally AssertionError, EOFError and Timeout(Error?) could be there for sure, so we might go for that, however we might lose some reliability with that 🤷

Reply:
I have slimmed the exceptions to `AssertionError` and `pexpect.exceptions.(EOF, TIMEOUT)` in the end.
Hopefully we won't miss anything :grimacing: